client/src/command/userSettings.ts

```diff
@@ -7,29 +7,30 @@ import {
     WorkspaceConfiguration,
     WorkspaceFolder
 } from 'vscode';
-import { IUserInterface, PickWithData } from '../IUserInterface';
+import { IUserInterface, Pick, PickWithData } from '../IUserInterface';
 import { VSCodeUI } from '../VSCodeUI';
 
 export async function setCheckstyleJar(ui: IUserInterface = new VSCodeUI()): Promise<void> {
-    await updateSetting(SettingType.JarPath, 'jarPath', ui);
+    const result: string = await ui.showFolderDialog({ Jar: ['jar'] });
+    await updateSetting('jarPath', result, ui);
 }
 
 export async function setCheckstyleConfig(ui: IUserInterface = new VSCodeUI()): Promise<void> {
-    await updateSetting(SettingType.ConfigPath, 'configPath', ui);
+    const browse: string = '$(file-directory) Browse...';
+    const configPicks: Pick[] = [
+        new Pick('google_checks'),
```

Comment:
Better to move "google_checks", "sun_checks" to a const var. Since they are referenced in multiple places.

Reply:
Sure

Reply:
will the ensurePath reject? if it could return reject, then there will be error.